demos/switch/js/SwitchDemo.js

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2017 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+var demo = demo || {};
+
+(function ($) {
+    "use strict";
+
+    demo.faces = {
+        primary: "😃",
+        random: ["😆", "😋", "😝", "😲"]
+    };
+
+    demo.switchPoint = 0.5;
+
+    demo.getFace = function () {
+        var rand = Math.random();
+
+        if (rand >= demo.switchPoint) {
+            demo.switchPoint += 0.05;
+            var faceIdx = Math.floor(Math.random() * (demo.faces.random.length));
+            return demo.faces.random[faceIdx];
+        } else {
+            return demo.faces.primary;
+        }
+    };
+
+    demo.toggleLight = function (state) {
+        var panel = $(".demo-panel");
```

Comment:
Shall this be written in the infusion way into a component such as `demo.panel`?